python/pyarrow/pandas_compat.py

```diff
@@ -61,15 +62,8 @@ def get_column_metadata(column, name):
 
     return {
         'name': name,
-        'pandas_type': {
-            'string': 'bytes' if PY2 else 'unicode',
-            'datetime64': (
-                'datetimetz' if hasattr(dtype, 'tz')
-                else 'datetime'
-            ),
-            'integer': str(dtype),
-            'floating': str(dtype),
-        }.get(inferred_dtype, inferred_dtype),
+        'arrow_type': str(arrow_type),
+        'pandas_type': str(pandas_type),
```

Comment:
I wonder if we should change the spec and use arrow types as logical types, otherwise I'm going to have to implement some kind of arrow type to pandas logical type which wouldn't be any better than just using the arrow type.

@wesm What do you think about that? I will put up the pandas PR.